akka-cluster-tools/src/main/scala/akka/cluster/client/ClusterClient.scala

```diff
@@ -628,6 +821,15 @@ final class ClusterReceptionist(pubSubMediator: ActorRef, settings: ClusterRecep
   val virtualNodesFactor = 10
   var consistentHash: ConsistentHash[Address] = ConsistentHash(nodes, virtualNodesFactor)
 
+  var clientInteractions = HashMap.empty[ActorRef, DeadlineFailureDetector]
+  var clientsPublished = HashSet.empty[ActorRef]
+
+  context.system.scheduler.schedule(
```

Comment:
I've learnt now that it is a requirement to kill off a scheduled task... I shall go and look at my other Akka code now!

Has there ever been thought given to having a local scheduler to an actor that automatically cancels tasks?